plum/overload.py

```diff
@@ -0,0 +1,21 @@
+import sys
+from typing import Any, Callable, TypeVar
+
+if sys.version_info >= (3, 11):  # pragma: specific no cover 3.7 3.8 3.9 3.10
+    from typing import get_overloads, overload
+else:  # pragma: specific no cover 3.11
+    from typing_extensions import get_overloads, overload
+
+from .function import Function
+
+__all__ = ["overload", "dispatch"]
+
+T = TypeVar("T", bound=Callable[..., Any])
+
+
+def dispatch(f: Callable) -> Function:
+    """Decorator to register a particular signature."""
+    f_plum = Function(f)
+    for method in get_overloads(f):
+        f_plum.dispatch(method)
+    return f_plum
```

Comment:
I think we can do an even better UX. What if, instead of using `from plum.overload import overload, dispatch` we could do `from plum import overload, dispatch`? This would mean that the `dispatch` function automatically detech overloads, which is possible in theory.

Can this logic be put in there: https://github.com/beartype/plum/blob/master/plum/dispatcher.py#L35 ?

The function signature is nearly the same, so it should work with mypy.

Something like:

```python

    def __call__(
        self,
        method: Optional[Callable] = None,
        precedence: int = 0,
    ) -> Callable:
        """Decorator to register for a particular signature.

        Args:
            precedence (int, optional): Precedence of the signature. Defaults to `0`.

        Returns:
            function: Decorator.
        """
        if method is None:
            return lambda m: self(m, precedence=precedence)

        overloads = get_overloads(method)
        if overloads != []:
            for overload_function in overloads:
                # the plum_function is always the same in this loop
                plum_function = self(overload_function)
            # method is never registered because it has no implementation
            return plum_function

        # The signature will be automatically derived from `method`, so we can safely
        # set the signature argument to `None`.
        return self._add_method(method, None, precedence=precedence)
```

Reply:
@gabrieldemarmiesse Ahhh this is clever. I've implemented it, and it works! Thank you for the awesome suggestion. :)